lab_4_fill_words_by_ngrams/main.py

```diff
@@ -98,6 +139,43 @@ def run(self, seq_len: int, prompt: str) -> str:  # type: ignore
                 or if sequence has inappropriate length,
                 or if methods used return None.
         """
+        if not isinstance(seq_len, int) or not isinstance(
+                prompt, str) or seq_len <= 0:
+            raise ValueError('TopPGenerator.run: inappropriate type of input arguments')
+
+        encoded = self._word_processor.encode(prompt)
+
+        if not encoded:
+            raise ValueError('TopPGenerator.run: encoded text is empty')
+
+        encoded_list = list(encoded)
+
+        for i in range(seq_len):
+            next_tokens = self._model.generate_next_token(tuple(encoded_list))
+
+            if next_tokens is None:
+                raise ValueError('TopPGenerator.run: generate_next_token returned None')
```

Comment:
You should see `TopPGenerator.run` on traceback. You can remove this prefic from messages

Reply:
done